src/relay/ir/text_printer.cc

```diff
@@ -277,16 +277,30 @@ class TextPrinter :
   TextValue VisitExpr_(const CallNode* op) final {
     // TODO(tqchen, M.K.): support generic call
     // possibly through meta-data
-    CHECK_EQ(op->type_args.size(), 0U)
-        << "generic call not yet supported";
     TextValue call_op = GetValue(op->op);
     std::vector<TextValue> args;
     for (Expr arg : op->args) {
       args.emplace_back(GetValue(arg));
     }
     TextValue id = this->AllocTempVar();
     this->PrintIndent();
-    stream_ << id << " = " << call_op << "(";
+
+    stream_ << id << " = " << call_op;
+
+    auto type_args = op->type_args;
```

Comment:
I am debating this change, since the type_infer populates the type_args field, it will make the code after type inference less concise. Can we lift the "Special Primitive Op detection" and not print this for special primitive ops? We had similar treatment for args

Reply:
Maybe we want a global option for whether or not to print these.

Reply:
for now they can be part of printer options